include/simdjson/dom/serialization.h

```diff
@@ -63,6 +64,9 @@ class string_builder {
 class mini_formatter {
 public:
   mini_formatter() = default;
+
+  virtual ~mini_formatter() = default;
```

Comment:
I'm OK with doing this another way, but I wonder if we can't just templatize all the places that print, and forego the virtual methods? Then the compiler would be able to inline everything, which we've seen over and over is one of the biggest performance factors.

I don't mind this for the pretty printer right now, though, as it's not part of simdjson's performance-focused features :)

Reply:
Done.